ngraph/core/src/op/pad.cpp

```diff
@@ -152,26 +152,23 @@ void op::v1::Pad::validate_and_infer_types()
         std::vector<Dimension> result_dims(implied_rank, Dimension::dynamic());
         for (size_t i = 0; i < implied_rank; i++)
         {
-            if (arg_shape[i].is_static())
+            result_dims[i] = arg_shape[i] + pads_begin_coord[i] + pads_end_coord[i];
```

Comment:
What if `arg_shape` has dynamic shape? Does dimension support + operator for dynamic?

Reply:
Yes